src/ukrdc_cupid/core/audit/validate_matches.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+from sqlalchemy.orm import Session
+import ukrdc_sqla.ukrdc as orm
+from sqlalchemy import select
+
+
+def validate_demog(session: Session, dob: datetime, pid: str) -> bool:
+    """Function to validate a date of birth against a patient record.
+
+    Args:
+        session (Session): ukrdc sqlachemy session
+        dob (datetime): incoming date of birth.
+        pid (str): persistant pid matched to incoming dob
+
+    Raises:
+        Exception: there should be only 1 (or possibly None) dob per patient
+        record. This exception needs fleshing see comment.
+
+    Returns:
+        bool: whether dob matches what we think it should
+    """
+
+    dob_query = select(orm.Patient.birthtime).where(orm.Patient.pid == pid)
```

Comment:
I'm (still?) unsure what this is doing as unless the PK/FKs have gone AWOL there will only ever be 1 result (or 0 if the PID doesn't exist).

Reply:
This function can be fleshed out basically its validation for a pid match. If you have particularly strong opinions about how it should be done differently I'm not particularly wedded how it currently works. It (should) return true if the persistent dob matches the incoming and produce an error if the query returns not 1.  Maybe this should be a workitem instead?

Reply:
I think I see what you're doing now.

I would probably do it as...

patient = session.get(Patient, pid)
if patient:
    if dob.date() != patient.BirthTime.date()
         raise Error
else:
          raise DifferentError